src/components/article-page.jsx

```diff
@@ -1,30 +1,50 @@
 import {useState, useEffect} from 'react'
-import { retriveIndividualArticles } from '../Api'
+import { retriveIndividualArticles, patchVotes } from '../Api'
 import {useParams} from 'react-router-dom'
 import CommentContainer from './comment-container'
 
 
+
 const ArticlePage = ({comments, setComments}) => {
     const [isLoading, setIsLoading] = useState(true)
     const [singleArticle, setSingleArticle] = useState()
+    const [votes, setVotes] = useState()
+    const [err, setErr] = useState (null)
     const id = useParams()
 
     useEffect(() => {
-        retriveIndividualArticles(id).then((article) => {
-            setSingleArticle(article)
+        retriveIndividualArticles(id).then(({article}) => {
+            setSingleArticle({article})
+            setVotes(article.votes)
             setIsLoading(false)
         })
     }, []) 
 
+    const handleVote = (vote) => {
+        setVotes((currVotes) => currVotes + vote)
+         setErr(null)
+          patchVotes(id, vote).catch((err) => {
+            setVotes((currVotes) => currVotes - vote)
+            setErr("Something went wrong, please try again!")
```

Comment:
Really nice work getting this to work for unsuccessful posts! The only thing I'd suggest is could this message maybe be displayed next to the voting buttons, so its easier to see?